### PR TITLE
Add some item props functionnality

### PR DESCRIPTION
**Pull request by Jhobean**
I have 3 bug on this version.

1-  (EDIT: FIXED)
Line 
`(pCharTarg->GetPropNum(pCCPChar, PROPCH_REFLECTPHYSICALDAM, pBaseCCPChar))`
always return 0... I set REFLECTPHYSICALDAM on target but dont know why always 0

2- (EDIT: I added LOS and range 5 to 10 depending of ERA)
is the damage area function. I copy pasted from 56d but after testing it it's just fucking non sense.
I hit everyone on the BIG screen including NPC,PET inside or outside house lol

3- (EDIT: I choose 3 type of damage link to physical)
How I can use only physical damage on iDmg ?

### CODE REVIEW

**Comment by drk84** [comment]
2)Take a look at the Spell_Area method, anyway this area damage properties should call onTakeDamage, don't make a new method onTakeDamageArea
3)The damage applied is based upon the utype flag

**Comment by Jhobean** [comment]
Seem OK on my side.
The only think i'm not sure is the Area damage hitting everyone.

Should I remove own pet? 
Should I remove all criminal action? How?

image

**Comment by Jhobean** [comment]
@drk84 I don't figure how avoid a new method for Takedamagearea...

Yes I use takeDamage on the new method.

What you think of this commit?

**Comment by cbnolok** [comment]
Did you figure out the last prop you were working on? Is it still a draft?

**Comment by Jhobean** [comment]
> Did you figure out the last prop you were working on? Is it still a draft?

Ready to merge. I fixed the "criminal act check"

File: src/game/chars/CCharFight.cpp [hunk 1]
```diff
@@ -939,8 +939,21 @@ int CChar::OnTakeDamage( int iDmg, CChar * pSrc, DAMAGE_TYPE uType, int iDmgPhys
...
 					}
-				}
-			}
+                }
+            }
+            // Check if REFLECTPHYSICALDAM will reflect some damage back.
+            // Preventing recurrent reflection with DAMAGE_REACTIVE.
+            if (!(uType & DAMAGE_REACTIVE))
+            {
+                int iReflectPhysical = minimum(GetPropNum(pCCPChar, PROPCH_REFLECTPHYSICALDAM, pBaseCCPChar),250); //Capped to 250
```
Comment (cbnolok) [inline src/game/chars/CCharFight.cpp:948]: Could you use std::min? Minimum is a macro and it calls two times GetPropNum, if you see how it's expanded

Comment (Jhobean) [reply src/game/chars/CCharFight.cpp]: Fixed. Good point!

**Comment by cbnolok** [comment]
Could you please fix the conflicts, in order to merge?

**Pull request merged**